pkg/smokescreen/stats_server.go

```diff
@@ -0,0 +1,84 @@
+package smokescreen
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+)
+
+type StatsServer struct {
+	config     *Config
+	ln         net.Listener
+	mux        *http.ServeMux
+	socketPath string
+}
+
+func newServer(config *Config) (s *StatsServer) {
+	s = &StatsServer{
+		config: config,
+		mux:    http.NewServeMux(),
+	}
+
+	s.mux.HandleFunc("/", s.stats)
+	return
+}
+
+func (s *StatsServer) Serve() {
+	pid := os.Getpid()
+	s.socketPath = fmt.Sprintf("%s/track-%d.sock", s.config.StatsSocketDir, pid)
+	ln, err := net.Listen("unix", s.socketPath)
+
+	if err != nil {
+		s.config.Log.Fatal("Could not start the reporting server.")
```

Comment:
I think we should include `err` in the message here.

Reply:
true!

Reply:
```{"level":"fatal","msg":"Could not start the reporting server.listen unix track/track-52727.sock: bind: no such file or directory","time":"2019-03-29T10:55:12-07:00"}```